src/ranged.cpp

```diff
@@ -573,17 +518,18 @@ int player::fire_gun( const tripoint &target, int shots, item& gun )
             use_charges( "UPS", gun.get_gun_ups_drain() );
         }
 
-        // Experience gain is limited by range and penalised proportional to inaccuracy.
-        int exp = std::min( range, 3 * ( get_skill_level( skill_used ) + 1 ) ) * 20;
-        int penalty = std::max( int( sqrt( shot.missed_by * 36 ) ), 1 );
-
-        // Even if we are not training we practice the skill to prevent rust.
-        practice( skill_used, train_skill ? exp / penalty : 0 );
 
         if( shot.missed_by <= .1 ) {
             lifetime_stats()->headshots++; // @todo check head existence for headshot
         }
 
+        if( shot.hit_critter && range > double( get_skill_level( skill_gun ) ) / MAX_SKILL * MAX_RANGE ) {
```

Comment:
This NEEDS to be revealed to player somehow. There is no way a typical ("expected value" of skill) player will make the connection.
For example, one in 10 shots at low range should comment on range being too low to train marksmanship.